app/controllers/ApplicationController.scala

```diff
@@ -387,4 +387,58 @@ trait ApplicationController extends BaseController with GatekeeperAuthWrapper {
     }
   }
 
+  def createPrivOrROPCApplicationPage(): Action[AnyContent] = { requiresRole(Role.APIGatekeeper, requiresSuperUser = true) {
+    implicit request =>
+      implicit hc => {
+        Future.successful(Ok(createPrivOrROPCApplication(createPrivOrROPCAppForm.fill(CreatePrivOrROPCAppForm(None, "", "", "")))))
+      }
+    }
+  }
+
+  def createPrivOrROPCApplicationAction(): Action[AnyContent] = {
+    requiresRole(Role.APIGatekeeper, requiresSuperUser = true) {
+      implicit request => implicit hc => {
+
+        def handleInvalidForm(form: Form[CreatePrivOrROPCAppForm]) = {
+          Future.successful(BadRequest(createPrivOrROPCApplication(form)))
+        }
+
+        def handleValidForm(form: CreatePrivOrROPCAppForm): Future[Result] = {
+
+          val env = if(appConfig.isExternalTestEnvironment) Environment.SANDBOX else Environment.PRODUCTION
+
+          def handleFormWithValidName: Future[Result] =
+            form.accessType.flatMap(AccessType.from) match {
+            case Some(accessType) => {
+              val collaborators = Seq(Collaborator(form.adminEmail, CollaboratorRole.ADMINISTRATOR))
+
+              applicationService.createPrivOrROPCApp(env, form.applicationName, form.applicationDescription, collaborators, AppAccess(accessType, Seq())) map {
+                case CreatePrivOrROPCAppSuccessResult(appId, appName, appEnv, clientId, totp, access) =>
+                  Ok(createPrivOrROPCApplicationSuccess(appId, appName, appEnv, Some(access.accessType), totp.production, clientId))
+              }
+            }
+          }
+
+          def handleFormWithInvalidName: Future[Result] = {
+            val formWithErrors = CreatePrivOrROPCAppForm.invalidAppName(createPrivOrROPCAppForm.fill(form))
+            Future.successful(BadRequest(createPrivOrROPCApplication(formWithErrors)))
+          }
+
+          def hasValidName(apps: Seq[ApplicationResponse]) = {
```

Comment:
This function is a little smelly - could we tidy it up a little bit?
The magic "PRODUCTION" strings feel wrong.

Maybe...

```
def hasValidName(apps: Seq[ApplicationResponse]) = env match {
            case Environment.PRODUCTION => !apps.exists(app => (app.deployedTo == Environment.PRODUCTION.toString) && (app.name == form.applicationName))
            case _ => true
          }
```